### PR TITLE
Author component reference template and writing guidelines 

### PR DESCRIPTION
**Pull request by chanmosq**
## Change summary

This PR creates a **Templates** section in **Contributing to O3DE Docs** -- a home for templates and supplemental writing guidelines to help contributors write common O3DE docs. 

I also created the first topic for this section: Writing a component reference document

### Submission Checklist:

* [x] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [x] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [x] **Consistency** - Does the content consistently follow the [style guide](https://o3de.org/docs/contributing/to-docs/style-guide/)?
* [x] **Help the user** - Does the documentation show the user something *meaningful*?

### CODE REVIEW

File: static/files/template-component-reference.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+
+## Provider
+
+[Gem's name](/docs/user-guide/gems/reference/<path-to-gem-docs>)
+
+
+## Dependencies
+
+<!-- (Optional) List the component's depdencies -->
```
Comment (willihay) [inline static/files/template-component-reference.md:19]:
```suggestion
<!-- (Optional) List the component's dependencies -->
```

File: static/files/template-component-reference.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+## Limitations
+
+<!-- (Optional) Write the limitations of this component. Don't include any "future work". -->
+
+
+## Properties
+
+![<component name> interface](/images/<path-to-images>)
+
+| Property | Description | Value | default |
```
Comment (willihay) [inline static/files/template-component-reference.md:36]:
```suggestion
| Property | Description | Value | Default |
```

File: static/files/template-component-reference.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+
+## <Component feature>
+
+<!-- (Optional) Explain specific features of your component. Repeat this section for each additional topic, as needed. -->
+
+
+## <EBus interface name>RequestBus
+
+<!-- (Optional) Example: "Use the following request functions with the <Request EBus name> EBus interface to communicate with <components> in your game." -->
```
Comment (willihay) [inline static/files/template-component-reference.md:50]:
```suggestion
<!-- (Optional) Example: "Use the following request functions with the <Request EBus name> EBus interface to communicate with <component name> in your game." -->
```

File: static/files/template-component-reference.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+| Request Name | Description | Paramter | Return | Scriptable |
```
Comment (willihay) [inline static/files/template-component-reference.md:52]: I've seen both "Request Name" and "Method Name" here. I prefer the more generic "Method Name", as seen in your Vegetation component example.
```suggestion
| Method Name | Description | Parameter | Return | Scriptable |
```

File: static/files/template-component-reference.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+| - | - | - | - | - |
+|   |   |   |   |   |
+|   |   |   |   |   |
+|   |   |   |   |   |
+
+## <EBus interface name>NotificationBus
+
+<!-- (Optional) Example: "Use the following notification functions with the <Notification EBus name> EBus interface to communicate with <components> in your game." -->
+
+| Request Name | Description | Paramter | Return | Scriptable |
```
Comment (willihay) [inline static/files/template-component-reference.md:62]:
```suggestion
| Method Name | Description | Parameter | Return | Scriptable |
```

File: content/docs/contributing/to-docs/templates/component-reference-doc.md [hunk 6]
```diff
@@ -0,0 +1,428 @@
...
+
+* Explain how users can work around a limitation, if possible. 
+
+* Write in complete sentences and elaborate on any technical terminology.
+
+* Don't include any "future work". A better place for that is the feature's roadmap, which can be found in the corresponding [SIG repository](https://www.o3de.org/contribute/#join-a-special-interest-group). 
+
+**Examples**:
+
+* [Physx Shape Collider component](/docs/user-guide/components/reference/physx/shape-collider/#limitations)
```
Comment (willihay) [inline content/docs/contributing/to-docs/templates/component-reference-doc.md:159]:
```suggestion
* [PhysX Shape Collider component](/docs/user-guide/components/reference/physx/shape-collider/#limitations)
```

File: static/files/template-component-reference.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+| - | - | - | - | - |
+|   |   |   |   |   |
+|   |   |   |   |   |
+|   |   |   |   |   |
+
+## <EBus interface name>NotificationBus
+
+<!-- (Optional) Example: "Use the following notification functions with the <Notification EBus name> EBus interface to communicate with <components> in your game." -->
```
Comment (willihay) [inline static/files/template-component-reference.md:60]:
```suggestion
<!-- (Optional) Example: "Use the following notification functions with the <Notification EBus name> EBus interface to communicate with <component name> in your game." -->
```

File: content/docs/contributing/to-docs/templates/component-reference-doc.md [hunk 8]
```diff
@@ -0,0 +1,428 @@
...
+* [Vegetation Altitude Filter component](/docs/user-guide/components/reference/vegetation-filters/vegetation-altitude-filter/)
+* [Shape Reference component](/docs/user-guide/components/reference/shape/shape-reference/)
+
+
+See `template-component-reference.md`:
+```md
+
+## <EBus interface name>RequestBus
+
+<!-- (Optional) Example: "Use the following request functions with the <Request EBus name> EBus interface to communicate with <components> in your game." -->
```
Comment (willihay) [inline content/docs/contributing/to-docs/templates/component-reference-doc.md:371]: To clarify that we mean to refer to *this* component, by name:
```suggestion
<!-- (Optional) Example: "Use the following request functions with the <Request EBus name> EBus interface to communicate with <component name> in your game." -->
```

File: content/docs/contributing/to-docs/templates/component-reference-doc.md [hunk 9]
```diff
@@ -0,0 +1,428 @@
...
+
+| Request Name | Description | Paramter | Return | Scriptable |
```
Comment (willihay) [inline content/docs/contributing/to-docs/templates/component-reference-doc.md:373]: See comment below about "Request" vs "Method". Also, a spelling error fix.
```suggestion
| Method Name | Description | Parameter | Return | Scriptable |
```

File: content/docs/contributing/to-docs/templates/component-reference-doc.md [hunk 10]
```diff
@@ -0,0 +1,428 @@
...
+| - | - | - | - | - |
+|   |   |   |   |   |
+|   |   |   |   |   |
+|   |   |   |   |   |
+
+## <EBus interface name>NotificationBus
+
+<!-- (Optional) Example: "Use the following notification functions with the <Notification EBus name> EBus interface to communicate with <components> in your game." -->
```
Comment (willihay) [inline content/docs/contributing/to-docs/templates/component-reference-doc.md:381]:
```suggestion
<!-- (Optional) Example: "Use the following notification functions with the <Notification EBus name> EBus interface to communicate with <component name> in your game." -->
```

File: content/docs/contributing/to-docs/templates/component-reference-doc.md [hunk 11]
```diff
@@ -0,0 +1,428 @@
...
+
+| Request Name | Description | Paramter | Return | Scriptable |
```
Comment (willihay) [inline content/docs/contributing/to-docs/templates/component-reference-doc.md:383]:
```suggestion
| Method Name | Description | Parameter | Return | Scriptable |
```

File: content/docs/contributing/to-docs/templates/component-reference-doc.md [hunk 12]
```diff
@@ -0,0 +1,428 @@
...
+
+* [Diffuse component](/docs/user-guide/components/reference/atom/diffuse-probe-grid/): A straightforward example with one dependency.
+
+* [Light component](/docs/user-guide/components/reference/atom/light/#dependencies): The Light component can depend on different Shape components. This example describes the varying dependency and its criteria.
+
+See `template-component-reference.md`:
+```md
+## Dependencies
+
+<!-- (Optional) List the component's depdencies -->
```
Comment (willihay) [inline content/docs/contributing/to-docs/templates/component-reference-doc.md:116]:
```suggestion
<!-- (Optional) List the component's dependencies -->
```

File: content/docs/contributing/to-docs/templates/component-reference-doc.md [hunk 13]
```diff
@@ -0,0 +1,428 @@
...
+|   |   |   |   |
+|   |   |   |   |
+|   |   |   |   |
+
+```
+
+{{% /tab %}}
+{{% tab name="Complex set of properties"%}}  
+
+For a component card with a complex set of properties, you may need multiple property tables: base properties, and one for each property group or configuration. The first property table is called "Base properties" and contains all of the properties that are displayed at the first level in the component card's content hierarchy. You can name subsequent property tables after the property group name in the component card or the type of configuration.
```
Comment (willihay) [inline content/docs/contributing/to-docs/templates/component-reference-doc.md:239]: You might want to mention when and how **tabs** might be needed in their property documentation, as shown in the Camera Rig component. Then you could link to the section on [tab shortcodes](https://o3de.org/docs/contributing/to-docs/style-guide/format/#tabs) in the style guide for details on how to create them in Markdown.

**Review by willihay — CHANGES_REQUESTED** [review]

File: content/docs/contributing/to-docs/templates/component-reference-doc.md [hunk 14]
```diff
@@ -0,0 +1,428 @@
...
+|   |   |   |   |   |
+
+```
+
+
+## Where does the component reference document live?
+
+Component reference documentation live in the folder: `o3de.org/docs/user-guide/components/reference/`. They are organized into subfolders that group the components by type. Your component reference document may belong to an existing group, or a new group completely. 
+
+Feel free to reach out to D&C SIG for assistance.
```
Comment (sptramer) [inline content/docs/contributing/to-docs/templates/component-reference-doc.md:396]:
```suggestion
Feel free to reach out to the [O3DE Documentation and Community Special Interest Group](https://github.com/o3de/sig-docs-community) for assistance. 
```

File: content/docs/contributing/to-docs/templates/component-reference-doc.md [hunk 15]
```diff
@@ -0,0 +1,428 @@
...
+## Where does the component reference document live?
+
+Component reference documentation live in the folder: `o3de.org/docs/user-guide/components/reference/`. They are organized into subfolders that group the components by type. Your component reference document may belong to an existing group, or a new group completely. 
+
+Feel free to reach out to D&C SIG for assistance. 
+
+
+## Storing image files
+
+The `o3de.org/static/images` folder is structured to reflect the `o3de.org/docs/` folder. So, you must store images for your component reference document in the appropriate component group folder: `o3de.org/static/images/user-guide/components/reference/<group>/`.
```
Comment (sptramer) [inline content/docs/contributing/to-docs/templates/component-reference-doc.md:401]: Recommend removing the `o3de.org` and using the repo structure here. 

```suggestion
The `static/images` directory is structured to reflect the `content/docs/` directory of the O3DE site repository. So, you must store images for your component reference document in the appropriate component group folder: `static/images/user-guide/components/reference/<group>/`.
```

File: content/docs/contributing/to-docs/templates/component-reference-doc.md [hunk 16]
```diff
@@ -0,0 +1,428 @@
...
+
+
+## Updating the O3DE Component Reference
+
+Remember to update the [Component Reference](/docs/user-guide/components/reference/) section and list your new component reference document. 
+
+
+1. In a text editor, open the file `o3de.org/docs/user-guide/components/reference.md`.
```
Comment (sptramer) [inline content/docs/contributing/to-docs/templates/component-reference-doc.md:409]:
```suggestion
1. In a text editor, open the file `content/docs/user-guide/components/reference.md` from your clone of the o3de.org repo.
```

File: content/docs/contributing/to-docs/templates/component-reference-doc.md [hunk 17]
```diff
@@ -0,0 +1,428 @@
...
+All documentation must adhere to the standards outlined in the O3DE [Style Guide](/docs/contributing/to-docs/style-guide/)and [Terminology](/docs/contributing/to-docs/terminology/). If you need any help, reach out to the Documentation and Community Special Interest Group (D&C SIG) on [Discord](https://discord.com/invite/o3de). 
+
+
+## Does my document belong in the O3DE Component Reference?
+
+First, determine whether or not your document belongs in the [Component Reference](/docs/user-guide/components/reference/) section of the O3DE User Guide. The purpose of the component reference document is exactly that, a page that users can quickly reference when using the component in **O3DE Editor**.
+
+Supplemental information, such as using the component across different workflows, belongs in other documentation. That document can be organized in an appropriate section of the User Guide, such as the [Interactivity and Simulation](/docs/user-guide/interactivity/), [Scripting](/docs/user-guide/scripting/), [Networking](/docs/user-guide/networking/), or [Visualization](/docs/user-guide/visualization/) sections.
+
+## Download template
```
Comment (sptramer) [inline content/docs/contributing/to-docs/templates/component-reference-doc.md:30]:
```suggestion
## Set up repository and download component template
```

Suggest that you add some info linking back to the instructions on setting up for contributing to o3de.org (https://www.o3de.org/docs/contributing/to-docs/o3de-docs-repo-setup/)

**Review by sptramer — CHANGES_REQUESTED** [review]
Some minor changes and suggestions. The template can be separated out and into the sig-docs-community repo afterwards.

File: static/files/template-component-reference.md [hunk 18]
```diff
@@ -0,0 +1,66 @@
+---
```
[the remark this answers is not in the document]
Comment (chanmosq) [reply static/files/template-component-reference.md:1]: Sounds good to me. I'll leave as is and fast follow with a PR to remove the template from the website, add it to o3de/sig-docs-community, and update the links.

File: content/docs/contributing/to-docs/templates/component-reference-doc.md [hunk 19]
```diff
@@ -0,0 +1,428 @@
...
+
+Like all Markdown (`.md`) files in the O3DE Documentation, your component reference document must contain metadata at the beginning of the file. Refer to [Metdata](/docs/contributing/to-docs/style-guide/metadata/) to help you fill in this section. 
+
+See `template-component-reference.md`:
+```yaml
+---
+linkTitle: <Component Name>
+title: <Component Name> Component
+description: The <Component Name>, which does <function>, is provided by the <Gem name> in Open 3D Engine (O3DE). # Example
+weight: 200 # Example
```
[the remark this answers is not in the document]
Comment (chanmosq) [reply content/docs/contributing/to-docs/templates/component-reference-doc.md]: I noted "optional" in the comment, and described the default behavior without weight in the guidelines.

File: content/docs/contributing/to-docs/templates/component-reference-doc.md [hunk 20]
```diff
@@ -0,0 +1,428 @@
...
+weight: 200
+toc: true
+---
+
+This topic guides you through the writing process for an **Open 3D Engine (O3DE)** component reference document.
+
+The workflow: 
+
+1. [Determine if your document belongs in the O3DE Component Reference](#does-my-document-belong-in-the-o3de-component-reference). 
+2. [Download the component reference document template.](#download-the-component-reference-document-template)
```
[the remark this answers is not in the document]
Comment (chanmosq) [reply content/docs/contributing/to-docs/templates/component-reference-doc.md]: I went with sptramer's suggestion but included the fix for consistent verb-pretense in the section titles. 
It now says ## Setting up repository and downloading component template.

File: content/docs/contributing/to-docs/templates/component-reference-doc.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (chanmosq) [reply content/docs/contributing/to-docs/templates/component-reference-doc.md]: Combined your and sptramer's suggestion, and changed Slack to Discord. 
It now says: "Feel free to reach out to sig-docs-community on [Discord](https://discord.com/invite/o3de) or [GitHub](https://github.com/o3de/sig-docs-community) for assistance."

File: static/files/template-component-reference.md [hunk 21]
```diff
@@ -0,0 +1,66 @@
...
+linkTitle: <Component Name>
+title: <Component Name> Component
+description: The <Component Name>, which does <function>, is provided by the <Gem name> in Open 3D Engine (O3DE). # Example
+weight: 200 # Example
```
[the remark this answers is not in the document]
Comment (chanmosq) [reply static/files/template-component-reference.md]: I noted "optional" in the comment, and described the default behavior without weight in the guidelines.

File: content/docs/contributing/to-docs/templates/component-reference-doc.md (hunk 13 above)
Comment (chanmosq) [reply content/docs/contributing/to-docs/templates/component-reference-doc.md:255]: Added!

File: static/files/template-component-reference.md [hunk 22]
```diff
@@ -0,0 +1,66 @@
...
+## Properties
+
+![<component name> interface](/images/<path-to-images>)
+
+| Property | Description | Value | default |
+| - | - | - | - |
+|   |   |   |   |
+|   |   |   |   |
+|   |   |   |   |
+
```
[the remark this answers is not in the document]
Comment (chanmosq) [reply static/files/template-component-reference.md]: Added! And updated the guidelines to describe use cases for tabs.

File: content/docs/contributing/to-docs/templates/component-reference-doc.md (hunk 14 above)
Comment (chanmosq) [reply content/docs/contributing/to-docs/templates/component-reference-doc.md]: Combined your and willihay's suggestion.
It now says: "Feel free to reach out to sig-docs-community on [Discord](https://discord.com/invite/o3de) or [GitHub](https://github.com/o3de/sig-docs-community) for assistance."

**Review by willihay — APPROVED** [review]

**Review by sptramer — APPROVED** [review]

**Pull request merged**